src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -159,7 +159,7 @@ protected static ITaskItem[] GenerateCommandLineArgsTaskItems(List<string> comma
             var items = new ITaskItem[commandLineArgs.Count];
             for (var i = 0; i < commandLineArgs.Count; i++)
             {
-                items[i] = new TaskItem(commandLineArgs[i]);
+                items[i] = new ArgsTaskItem(commandLineArgs[i]);
```

Comment:
This will change from us unconditionally escaping to unconditionally not escaping. Won't this regress scenarios where escaping is making builds work today? Consider for example that `/keyfile:` is passed via `commandLineArgs`. If that was using `\` today it would be `/` on unix but after this change it woudl still be `\` correct?

Reply:
oof. Luckily many args such as Analyzers, References, Source all come from TaskItems of their own which apply the path correction. So I need to look at args which come from properties and may contain file paths.

Reply:
So is the expectation that MSBuild project files might do things like <SomeProperty>Foo\Bar</SomeProperty> and this conversion is what makes things still work on non-Windows? That seems terrifying, but I guess project files using \ to make paths is _very_ common...